pgmpy/factors/CPD.py

```diff
@@ -107,27 +107,25 @@ def __init__(self, variable, variable_card, values,
 
         cardinality = [variable_card]
         if evidence_card is not None:
-            if not isinstance(evidence_card, (list, set, tuple)):
+            if not isinstance(evidence_card, (list, tuple)):
                 if isinstance(evidence_card, np.ndarray):
                     evidence_card = evidence_card.tolist()
                 elif isinstance(evidence_card, (int, float)):
                     evidence_card = [evidence_card]
                 else:
-                    raise TypeError("Evidence cardinality must be list, set, "
+                    raise TypeError("Evidence cardinality must be list, "
                                     "tuple, numpy ndarray or simply a number.")
-            self.evidence_card = evidence_card
             cardinality.extend(evidence_card[::-1])
```

Comment:
I have one more doubt. Should we allow reducing or marginalizing the variable itself in CPD, especially inplace. If it were a JPD then marginalizing wrt any variable would be correct, but I suspect that it is not the case here. 
Currently reducing and marginalizing w.r.t. `variable` is allowed and it gives all `1.` output as can be expected if this behavior is to be allowed. But, `print(cpd)` doesn't work as can also be expected. 
But, in my view this should not be allowed inplace. If inplace is chosen to `False` then it can be allowed I guess. 
Also, docstrings for marginalize and reduce are not correct, errors pop up if you follow the directions exactly.